docs/dynamic-programming/painting-fence.mdx

```diff
@@ -0,0 +1,142 @@
+---
+title: Painting Fence
+slug: /mathematics/painting-fence
+---
+
+## Introduction
+
+Given a fence with `n` posts and `k` colors , find out the number of ways of painting the fence such that atmost `2` adjacent posts have the same color.
+
+## Explanation
+
+Suppose we have a single post, The number of ways of painting the fence with `k` colors is `k` Now we consider `2` posts, Then the number of ways of painting the second fence with the same color as the first post is `k`, and the number of ways of painting the second post with a different color is `k*(k-1)` as we have `k-1` colors that are different from the first post's color. So the total number of ways of painting `2` fences is `k*(k-1) + k`. Now we consider `3` posts, Then the number of ways of painting the third fence same as the second fence is the number of ways of painting the second fence with a different color that is `k*(k-1)`. The number of ways of painting the third fence using a different color is equal to the `total number of ways of painting the second fence * (k-1)`. Hence the total number of ways of painting `3` fences is `k*(k-1) + (k+k*(k-1))*(k-1)`. This way we can find the total number of ways of painting n fences
```

Comment:
Better break this down into consumable bits. Also don't use back-tick formatting indiscriminately